### PR TITLE
Docker micro services implementation.

### PR DESCRIPTION
**Pull request by onyxim**
I completed a bunch of changes to move project for using docker micro services architecture. Also, I filled Readme.md with info about how to use it.

Instapy.py is completely backward compatible with old "local" selenium setup.

There is no more longer need to have old docker file. But I saved it for now. You can purge it later. Actually old docker file simply didn't builld. I tested it.

New docker files uses only official, trusted images from docker hub. Selenium images isn't supported by docker team, but by selenium official developers.

I use PyCharm for develop and testing InstaPy, so docker_dev.yml makes a container to access with ssh for a remote debug. So, I think it'll be helpful for other IDEs.

### CODE REVIEW

File: instapy/instapy.py [hunk 1]
```diff
@@ -81,6 +74,26 @@ def __init__(self, username=None, password=None, nogui=False):
 
         self.aborting = False
 
+    def set_selenium_local_session(self):
+        chromedriver_location = './assets/chromedriver'
```
Comment (timgrossmann) [inline instapy/instapy.py]: Make sure to add the same behaviour as with the other methods:

Use the `if self.aborting...` and the `return self` to allow for method chaining.
Check the other methods to see what I mean

File: instapy/instapy.py [hunk 2]
```diff
@@ -81,6 +74,26 @@ def __init__(self, username=None, password=None, nogui=False):
...
+        chrome_options.add_argument('--lang=en-US')
+        chrome_options.add_experimental_option('prefs', {'intl.accept_languages': 'en-US'})
+        chrome_options.binary_location = chromedriver_location
+        self.browser = webdriver.Chrome(chromedriver_location, chrome_options=chrome_options)
+        self.browser.implicitly_wait(25)
+        self.logFile.write('Session started - %s\n' \
+                           % (datetime.now().strftime('%Y-%m-%d %H:%M:%S')))
+
+    def set_selenium_remote_session(self, selenium_url=''):
+        self.browser = webdriver.Remote(command_executor=selenium_url,
```
Comment (timgrossmann) [inline instapy/instapy.py]: Same here 😉

**Comment by onyxim** [comment]
I amended the last commit with new changes. Just check it.

**Comment by timgrossmann** [comment]
@onyxim Actually you're right with the docker part... This would remove a lot of problems.

Well, method chaining is just really nice and pretty 😄

**Comment by timgrossmann** [comment]
@onyxim Are you part of the discord channel? 
@scriptworld-git would like to ask you some questions in regards to the docker files in the repository since we want to completely extract the docker part out to this repository: https://github.com/InstaPy/instapy-docker

Thank you

**Pull request merged**